kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyReconciler.java

```diff
@@ -296,9 +297,33 @@ public Map<String, EventSource> prepareEventSources(EventSourceContext<KafkaProx
                         e);
             }
         }
+        eventSources.add(buildVirtualKafkaClusterInformer(context));
         return EventSourceInitializer.nameEventSources(eventSources.toArray(new EventSource[0]));
     }
 
+    private static InformerEventSource<?, KafkaProxy> buildVirtualKafkaClusterInformer(EventSourceContext<KafkaProxy> context) {
+        InformerConfiguration<VirtualKafkaCluster> configuration = InformerConfiguration.from(VirtualKafkaCluster.class).withSecondaryToPrimaryMapper(resource -> {
+            // we need to reconcile all proxies when a virtual kafka cluster changes in case the proxyRef is updated, we need to update
+            // the previously referenced proxy too.
+            Set<ResourceID> proxyIds = context.getClient().resources(KafkaProxy.class)
+                    .inNamespace(resource.getMetadata().getNamespace())
+                    .list().getItems().stream()
+                    .map(ResourceID::fromResource)
+                    .collect(Collectors.toSet());
```

Comment:
nit: The mix of chop down on method call and multiple calls on a single line (310) is a bit jarring

Reply:
refactored it a lot, covers several of your nits.